server/mail/mail.go

```diff
@@ -78,7 +78,15 @@ func getMessageBody(e fleet.Email, f fromFunc) ([]byte, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to obtain from address: %w", err)
 	}
-	msg := []byte(subject + from + mime + content + "\r\n" + string(body) + "\r\n")
+	to := ""
+	if len(e.To) == 1 {
+		to = "To: " + e.To[0] + "\r\n"
+	} else if len(e.To) > 1 {
+		// Splitting the header into multiple lines as a precaution
+		// https://www.rfc-editor.org/rfc/rfc2822.html#section-2.2.3
+		to = "Bcc: " + strings.Join(e.To, ",\r\n    ") + "\r\n"
```

Comment:
We don't want the BCC to be in the text of the header sent here, but rather the `to` to say something like `"To: undisclosed-recipients\r\n"` or the like.

BCC can be in the envelope, but we don't want it in the message headers.

Reply:
Gotcha 👍